virttest/qemu_vm.py

```diff
@@ -1055,6 +1055,13 @@ def add_memorys(devices, params):
                 backend_options["size_mem"] = "%sM" % params["mem"]
                 if params.get("vm_mem_backend"):
                     backend_options["backend_mem"] = params.get("vm_mem_backend")
+                if params.get("vm_mem_backend") == "memory-backend-file":
+                    if not params.get("vm_mem_backend_path"):
+                        raise virt_vm.VMConfigMissingError(name,
+                                                           'vm_mem_backend_path')
+                    else:
+                        backend_options["mem-path_mem"] = params[
+                            "vm_mem_backend_path"]
```

Comment:
After a basic check, I think the current code can capture the error info but cannot clearly raise it: `ERROR: VM 'avocado-vt-vm1' failed to start: Error occurred while executing make_cre
ate_command(). Check the log for traceback.`
```
[stderr] Traceback (most recent call last):
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/qemu_vm.py", line 3317, in create
[stderr]     self.devices, self.spice_options = self.make_create_command()
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/qemu_vm.py", line 1773, in make_create_command
[stderr]     add_memorys(devices, params)
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/qemu_vm.py", line 1097, in add_memorys
[stderr]     raise virt_vm.VMConfigMissingError(name,
[stderr] virttest.virt_vm.VMConfigMissingError: Missing config 'vm_mem_backend_path' for VM machine_mem
[stderr]
[stderr] During handling of the above exception, another exception occurred:
[stderr]
[stderr] Traceback (most recent call last):
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/avocado_vt/plugins/vt_runner.py", line 71, in runTest
[stderr]     raise self.__status  # pylint: disable-msg=E0702
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/avocado_vt/test.py", line 145, in setUp
[stderr]     self._runTest()
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/avocado_vt/test.py", line 252, in _runTest
[stderr]     params = env_process.preprocess(self, params, env)
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/error_context.py", line 135, in new_fn
[stderr]     return fn(*args, **kwargs)
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/env_process.py", line 1383, in preprocess
[stderr]     process(test, params, env, preprocess_image, preprocess_vm,
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/env_process.py", line 889, in process
[stderr]     _call_vm_func()
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/env_process.py", line 801, in _call_vm_func
[stderr]     vm_func(test, vm_params, env, vm_name)
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/env_process.py", line 300, in preprocess_vm
[stderr]     vm.create(name, params, test.bindir,
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/error_context.py", line 135, in new_fn
[stderr]     return fn(*args, **kwargs)
[stderr]   File "/home/yihyu/kar/workspace/avocado-vt/virttest/qemu_vm.py", line 3330, in create
[stderr]     raise virt_vm.VMStartError(self.name, 'Error occurred while '
[stderr] virttest.virt_vm.VMStartError: VM 'avocado-vt-vm1' failed to start: Error occurred while executing make_create_command(). Check the log for traceback.
```
`self.create()` will handle the exception: https://github.com/avocado-framework/avocado-vt/blob/e20d6f47c2198f423bbc8f420b2775e404322416/virttest/qemu_vm.py#L3319-L3325

However, `utils_params` can natively handle missing parameters, and make the case status clearer: `SKIP: Mandatory parameter 'vm_mem_backend_path' is missing. Check your cfg files for typos/mistakes`
```suggestion
                if params.get("vm_mem_backend") == "memory-backend-file":
                    backend_options["mem-path_mem"] = params[
                        "vm_mem_backend_path"]
```

What do you think, guys?

Reply:
Hi @PaulYuuu 
I prefer to keep the current way to raise errors since I think the `SKIP` result does not belong to the error, it will result in collecting and counting the test case pass ratio..
And we also get the some hits from the last error info : `Error occurred while executing make_create_command(). Check the log for traceback.` which will lead us to get the more info by checking the log for traceback.
Thanks.

Reply:
Same with @YongxueHong , it is an auto issue an need be fixed, `SKIP` status is not make sense IMO.